kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -290,101 +277,103 @@ object Result:
 
         /** Gets the panic exception if present.
           *
           * @return
           *   A Maybe containing the panic exception, or empty for non-Panic Results
           */
-        @targetName("maybePanic")
-        def panic: Maybe[Throwable] =
-            self match
-                case self: Panic => Maybe(self.exception)
-                case _           => Maybe.empty
+        def panic: Maybe[Throwable] = foldAll(Present(_))(_ => Absent)(_ => Absent)
 
-        /** Folds the Result into a value.
+        /** Gets either the failure value or panic exception if present.
           *
-          * @param ifFailure
-          *   Function to apply if the Result is a failure
+          * @return
+          *   A Maybe containing either the failure value or panic exception, or empty for Success Results
+          */
+        def failureOrPanic: Maybe[E | Throwable] = foldAll(Present(_))(Present(_))(_ => Absent)
+
+        /** Folds a Result into a value, with separate handling for errors and successes.
+          *
+          * @param ifError
+          *   Function to apply if the Result is an Error (either Failure or Panic)
           * @param ifSuccess
-          *   Function to apply if the Result is a success
-          * @tparam B
-          *   The type of the result of folding
+          *   Function to apply if the Result is a Success
           * @return
           *   The result of applying the appropriate function
           */
-        inline def fold[B](inline ifFailure: Error[E] => B)(inline ifSuccess: A => B): B =
-            self match
-                case self: Error[E] @unchecked => ifFailure(self)
-                case _ =>
-                    try ifSuccess(self.asInstanceOf[Result[Nothing, A]].get)
-                    catch
-                        case ex => ifFailure(Panic(ex))
+        inline def foldError[B](ifError: Error[E] => B)(inline ifSuccess: A => B): B =
+            try
+                self match
+                    case error: Error[E] @unchecked => ifError(error)
+                    case _                          => ifSuccess(self.asInstanceOf[Result[Nothing, A]].getOrThrow)
+            catch
+                case ex => ifError(Panic(ex))
 
-        /** Gets the successful value.
+        /** Folds a Result into a value, with separate handling for failures and successes, throwing any panics.
           *
-          * @param ev
-          *   Evidence that E is Nothing
+          * @param ifFailure
+          *   Function to apply if the Result is a Failure
+          * @param ifSuccess
+          *   Function to apply if the Result is a Success
           * @return
-          *   The successful value
-          * @throws NoSuchElementException
-          *   if the Result is a Fail
+          *   The result of applying the appropriate function
           * @throws Throwable
           *   if the Result is a Panic
           */
-        def get(
```

Comment:
It didn't make much sense to have both `get` and `getOrThrow`. I've kept only `getOrThrow` since it indicates that it may throw